plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -122,7 +113,16 @@ public function add_navigation_id_directive( $block_content, $block, $instance )
 					'data-wc-navigation-id',
 					'wc-product-collection-' . $this->parsed_block['attrs']['queryId']
 				);
-				$p->set_attribute( 'data-wc-interactive', wp_json_encode( array( 'namespace' => 'woocommerce/product-collection' ) ) );
+				$p->set_attribute( 'data-wc-interactive', '{ "namespace": "woocommerce/product-collection" }' );
```

Comment:
Can you explain why we don't need that encoding anymore?

Reply:
I aimed to maintain consistency with the syntax used in the Query Loop block, as you can observe [here](https://github.com/WordPress/gutenberg/blob/621a3cd0ee1b86e0a213e820a159fac033cb7a7b/packages/block-library/src/query/index.php#L30). Do you foresee any issues with this approach?

CC: @DAreRodz, Was there a specific reason for the syntax in the query loop block to be different?

Reply:
Also, after implementing that change, everything is functioning as anticipated.

Reply:
There is no difference between any of the approaches; both render the same JSON code. So, there's no specific reason other than avoiding potential mistakes while manually writing a JSON data structure. 😄

Reply:
> Do you foresee any issues with this approach?

No, like I wrote in main comment, the questions "are only for my education and do not intend to challenge anything." 😆 I have trust in what's happening in Query Loop and your assessment so don't worry too much about my questions!